### ISSUE TITLE
Order of Categorical Labels in explainer.expected_value 

### ISSUE THREAD

**Issue by msarafzadeh** [shap/shap#1117]
What is the order of the labels in explainer.expected_value ?
I am using pything with XGBoost Classifier with categorical variables as my target.

Set and list have different orders (set really doesn't have an 'order').

![image](https://user-images.githubusercontent.com/12986465/77319683-ed3b8e00-6d17-11ea-8c0e-eb4a041b7d1c.png)

**Comment by alexcoca** [comment]
I gather from the above that the dimensionality of the model output is 3. Each logit in your output corresponds to one of your classes, depending on what the order of your target vector during training. `shap` preserves that order.

**Comment by msarafzadeh** [comment]
If the target vector is a mix of different categories, what order is it then?

**Comment by alexcoca** [comment]
Can you give a description of your training data? What are the inputs to the model and what are your targets, together with the data structures you use for them?

**Comment by msarafzadeh** [comment]
inputs are numerical, outputs are a single target of type ['Yes','No','Maybe']

**Comment by alexcoca** [comment]
So there are 3 categories involved, and for each we aim to predict a Yes/No/Maybe response. So an example in each of the categories the target would be [Yes, Yes, Yes]?

**Comment by msarafzadeh** [comment]
or for example [Yes,No,Yes,Yes,Yes,Maybe,No]....

**Comment by alexcoca** [comment]
I don't think I understand your setup. I expect the targets to have the same dimensionality across examples.

**Comment by msarafzadeh** [comment]
It's one target per sample. There are 3 possible target categorical types.
The SHAP has an expected value calculated for each target. However, what is
the order of the targets for the SHAPs.

**Comment by alexcoca** [comment]
So your model outputs one scalar?

**Comment by alexcoca** [comment]
I would expect it's the same as implied for your model. If your model predicts vector of 3 numbers, one for each categorical type, where the first number is the log-odds for 'Yes', second for 'No' and 3rd for 'Maybe', then the first number in the expected value vector is for 'Yes', second for 'No', third for 'Maybe'.

**Comment by msarafzadeh** [comment]
it's a XGBClassifier model. It has only one target output.

**Comment by msarafzadeh** [comment]
I am not sure what the order is. How do you know if that's the case? Is
the order lexical  by category name?
